test/Microsoft.ML.Tests/Scenarios/Api/Estimators/DeserializationTests.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.ML.Calibrators;
+using Microsoft.ML.Data;
+using Microsoft.ML.RunTests;
+using Microsoft.ML.Trainers.FastTree;
+using Xunit;
+
+namespace Microsoft.ML.Tests.Scenarios.Api
+{
+    public partial class ApiScenariosTests
+    {
+        private class InputData
+        {
+            [LoadColumn(0)]
+            public float Label { get; set; }
+            [LoadColumn(9, 14)]
+            [VectorType(6)]
+            public float[] Features { get; set; }
+        }
+
+        [Fact]
+        public void LoadModelAndExtractPredictor()
+        {
+            var ml = new MLContext(seed: 1, conc: 1);
+            var file = new MultiFileSource(GetDataPath(TestDatasets.adult.trainFilename));
+            var loader = ml.Data.CreateTextLoader<InputData>(hasHeader: true, dataSample: file);
+            var data = loader.Load(file);
+
+            // Pipeline.
+            var pipeline = ml.BinaryClassification.Trainers.GeneralizedAdditiveModels();
+
+            // Train.
+            var model = pipeline.Fit(data);
+
+            // Save and reload.
+            string modelPath = GetOutputPath(FullTestName + "-model.zip");
+            using (var fs = File.Create(modelPath))
+                ml.Model.Save(model, fs);
+
+            ITransformer loadedModel;
+            using (var fs = File.OpenRead(modelPath))
+                loadedModel = ml.Model.Load(fs);
+
+            var gam = (((loadedModel as TransformerChain<ITransformer>).LastTransformer
+                as BinaryPredictionTransformer<object>).Model
+                as CalibratedModelParametersBase).SubModel
+                as BinaryClassificationGamModelParameters;
+            Assert.NotNull(gam);
+        }
+
+        [Fact]
+        public void SaveAndLoadModelWithLoader()
```

Comment:
Could we also have a test for the scenario where the loader is part of the pipeline and we try to save the entire pipeline? Something that looks like the following:

```csharp
var pipeline = mlContext.Data.CreateTextLoader(....)
                        .Append(mlContext.Transforms.Conversions.Hash(...)
                        .Append(....)

var model = pipeline.Fit(stream);

mlContext.Model.Save(model, fs);
```

Reply:
Would be good to make sure we can then use `LoadAsCompositeDataLoader`.

---
In reply to: [263153288](https://github.com/dotnet/machinelearning/pull/2858#discussion_r263153288) [](ancestors = 263153288)